### PR TITLE
Destructure scanner methods in parser

### PR DESCRIPTION
**Pull request by sandersn**
This might be faster!

Addresses https://github.com/microsoft/TypeScript/issues/52959

### CODE REVIEW

**Comment by DanielRosenwasser** [comment]
@typescript-bot perf test this

**Comment by sandersn** [comment]
Local run on node 19 with `--predictable` shows no changes in overall time except for TFS, which gets 1% faster (and nearly every stage's timings jump around). xstate's parsing gets 1% faster too, which indicates to me that all those `scanner.getTokenPos` calls in the JSDoc parsing were costing us something.

If the perf machine agrees, this is a good change even if it's only a small bump.

**Comment by sandersn** [comment]
The failing failed to download the action at all. I don't think it's relevant in any case.

File: src/compiler/parser.ts [hunk 1]
```diff
@@ -1407,7 +1407,51 @@ namespace Parser {
...
+        getStartPos,
+        getTextPos,
+        getToken,
+        getTokenFlags,
+        getTokenPos,
+        getTokenPos,
+        getTokenText,
+        getTokenValue,
+        hasExtendedUnicodeEscape,
+        hasExtendedUnicodeEscape: scanner_hasExtendedUnicodeEscape,
```
Comment (jakebailey) [inline src/compiler/parser.ts:1423]: Is it bad that I wish these all were `scanner_` prefixed? Maybe it's not such a big deal to remember where these come from, I suppose.

**Review by jakebailey — COMMENTED** [review]
Is there some subset of these we could inline instead of doing all of them, sorta like we did factories?

**Comment by jakebailey** [comment]
Eek, it's worse?

**Comment by sandersn** [comment]
`--predictable` gives different results, but they are very small, so I don't want to bet that the normal perf results are wrong and my --predictable local run is right.

**Pull request closed without merging**